File: joi2008yo/e/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int R, C;

int main() {
  cin >> R >> C;
  vector<vector<int>> state(R, vector<int>(C));
  for (int i = 0; i < R; ++i) {
    for (int j = 0; j < C; ++j) {
      cin >> state[i].at(j);
    }
  }
  int res = 0;
  for (int r = 0; r < (1 << R); ++r) {
    int sum = 0;
    for (int i = 0; i < C; ++i) {
      int front = 0;
      for (int j = 0; j < R; ++j) {
        int v = state[j][i];
        if (r & (1 << j)) v = v ^ 1;
        if (v) front += 1;
      }
      sum += max(front, R - front);
    }
    if (sum > res) res = sum;
  }

  cout << res << endl;
}

File: joi2008yo/e/main.rb
R, C = gets.split.map(&:to_i)
state = R.times.map { gets.chomp.split.map(&:to_i) }

def max(a, b); a > b ? a : b; end

ans = 0
(0...(1 << R)).each do |r_choice|
  sum = 0
  C.times do |i| # col
    front = 0
    R.times do |j| # row
      v = state[j][i] ^ r_choice[j]
      front += 1 if v == 1
    end
    sum += max(front, R - front)
  end
  ans = sum if sum > ans
end
puts ans